src/packages/utils/src/things/json-rpc/json-rpc-data.ts

```diff
@@ -1,57 +1,91 @@
 import { BaseJsonRpcType } from "./json-rpc-base-types";
-import { strCache, toStrings } from "./json-rpc-base-types";
+import { BUFFER_EMPTY } from "../../utils/constants";
 
-function validateByteLength(byteLength?: number) {
-  if (typeof byteLength !== "number" || !(byteLength >= 0)) {
-    throw new Error(`byteLength must be a number greater than or equal to 0, provided: ${byteLength}`);
-  }
-}
+export type JsonRpcDataInputArg = string | Buffer;
 
 export class Data extends BaseJsonRpcType {
+  public static Empty = Data.from(BUFFER_EMPTY);
 
-  constructor(value: string | Buffer, private _byteLength?: number) {
+  constructor(value: JsonRpcDataInputArg, private _byteLength?: number) {
```

Comment:
That second case _would_ pass, but it would be expected to, as `Buffer.from(string)` will use utf8 encoding by default - so we end up passing a byte array into the `Data` constructor.